{
  "rustsec_label": "Safe",
  "in_rustsec_patched": false,
  "in_rustsec_current": false,
  "in_rust_sec": false,
  "rustsec_tag": null,
  "developers": [
    "termoshtt",
    "kngwyu",
    "github:rust-numpy:publish",
    "github:pyo3:rust-numpy"
  ],
  "stars": 1208,
  "forks": 120,
  "downloads": 6676675,
  "num_side_effects": 221,
  "failed_rudra": false,
  "audits": [
    {
      "organization": "google",
      "criteria": "ub-risk-4",
      "delta": "0.20.0 -> 0.21.0",
      "version": "",
      "notes": "Reviewed in CL 683848897\\nIssues found:\\n - to_owned_array needs to be unsafe as it can introduce aliasing UB\\n - Review incomplete: pervasive undocumented unsafety\\n"
    }
  ],
  "dependencies": [
    [
      "libc",
      "0.2.171"
    ],
    [
      "ndarray",
      "0.16.1"
    ],
    [
      "matrixmultiply",
      "0.3.9"
    ],
    [
      "rawpointer",
      "0.2.1"
    ],
    [
      "num-complex",
      "0.4.6"
    ],
    [
      "num-traits",
      "0.2.19"
    ],
    [
      "num-integer",
      "0.1.46"
    ],
    [
      "portable-atomic",
      "1.11.0"
    ],
    [
      "portable-atomic-util",
      "0.2.4"
    ],
    [
      "pyo3",
      "0.24.0"
    ],
    [
      "cfg-if",
      "1.0.0"
    ],
    [
      "memoffset",
      "0.9.1"
    ],
    [
      "once_cell",
      "1.21.1"
    ],
    [
      "pyo3-ffi",
      "0.24.0"
    ],
    [
      "rustc-hash",
      "2.1.1"
    ]
  ],
  "passed_audit": false,
  "num_unsafe_calls": 196,
  "miri": true,
  "past_audit": true,
  "miri_details": {
    "status": "crash",
    "passed": 0,
    "failed": 3,
    "ignored": 0,
    "measured": 0,
    "filtered_out": 0,
    "time_seconds": 0.0
  }
}